src/classes/CW_AssociateAudienceRewardHelperTest.cls

```diff
@@ -0,0 +1,124 @@
+		System.AssertEquals(3, getBoolCount(ctrHlpr.getCheckkAgsr().Values()));
+		ctrHlpr.PresistChages();
+		System.AssertEquals(15, database.countquery('SELECT COUNT()  FROM CW_Audience_Group_Schedule__c ')); 	
+		System.AssertEquals(3, database.countquery('SELECT COUNT()  FROM CW_Audience_Group_Schedule_Reward__c '));
+		
+		ctrHlpr.contextItem=''+shedLst[0].Id+listGroups[0].id+listRewards[0].id;
+		ctrHlpr.doSelectGroupReward();
+		ctrHlpr.contextItem=''+shedLst[1].Id+listGroups[1].id+listRewards[1].id;
+		ctrHlpr.doSelectGroupReward();
+		ctrHlpr.contextItem=''+shedLst[2].Id+listGroups[2].id+listRewards[2].id;
+		ctrHlpr.doSelectGroupReward();
+		System.AssertEquals(4, getBoolCount(ctrHlpr.getCheckkAgs().Values()));
+		System.AssertEquals(6, getBoolCount(ctrHlpr.getCheckkAgsr().Values()));
+		ctrHlpr.PresistChages();
+		System.AssertEquals(15, database.countquery('SELECT COUNT()  FROM CW_Audience_Group_Schedule__c ')); 	//all posible scheduled
+		System.AssertEquals(6, database.countquery('SELECT COUNT()  FROM CW_Audience_Group_Schedule_Reward__c '));
+		
+		ctrHlpr= new CW_AssociateAudienceRewardHelper(camp.Id);
+		
+		//Recheck 
+		ctrHlpr.contextItem=''+shedLst[0].Id+listGroups[0].id+listRewards[0].id;
+		ctrHlpr.doDeSelectGroupReward();
+		ctrHlpr.contextItem=''+shedLst[0].Id+listGroups[0].id+listRewards[0].id;
+		ctrHlpr.doSelectGroupReward();
+		System.AssertEquals(4, getBoolCount(ctrHlpr.getCheckkAgs().Values()));
+		System.AssertEquals(6, getBoolCount(ctrHlpr.getCheckkAgsr().Values()));
+		
+		ctrHlpr.contextItem=''+shedLst[1].Id+listGroups[2].id+listRewards[0].id;
+		ctrHlpr.doSelectGroupReward();
+		System.AssertEquals(5, getBoolCount(ctrHlpr.getCheckkAgs().Values()));
+		System.AssertEquals(7, getBoolCount(ctrHlpr.getCheckkAgsr().Values()));
+		
+		ctrHlpr.contextItem=''+listGroups[2].id+listRewards[0].id;
+		ctrHlpr.doDeSelectGroupSchedule();
+		System.AssertEquals(4, getBoolCount(ctrHlpr.getCheckkAgs().Values()));
+		System.AssertEquals(6, getBoolCount(ctrHlpr.getCheckkAgsr().Values()));
+		
+		ctrHlpr.contextItem=''+listGroups[0].id+listRewards[0].id;
+		ctrHlpr.doDeSelectGroupSchedule();
+		System.AssertEquals(5, getBoolCount(ctrHlpr.getCheckkAgsr().Values()));
+		
+		//Delete records
+		ctrHlpr.contextItem=''+shedLst[0].Id+listGroups[0].id+listRewards[0].id;
+		ctrHlpr.doDeSelectGroupReward();
+		ctrHlpr.contextItem=''+listGroups[0].id+listRewards[1].id;
+		ctrHlpr.doDeSelectGroupSchedule();
+		System.AssertEquals(4, getBoolCount(ctrHlpr.getCheckkAgs().Values()));
+		System.AssertEquals(2, getBoolCount(ctrHlpr.getCheckkAgsr().Values()));
+		ctrHlpr.PresistChages();
+		//No delete		
+		System.AssertEquals(15, database.countquery('SELECT COUNT()  FROM CW_Audience_Group_Schedule__c ')); 	
+		System.AssertEquals(2, database.countquery('SELECT COUNT()  FROM CW_Audience_Group_Schedule_Reward__c '));
+	}
+   
+   	private static integer getBoolCount(List<Boolean> bvl)
+   	{
+		integer i=0;
```

Comment:
This variable doesn't seem to have a very meaningful name. I suggest to give this variable a concise but meaningful name that suggest its purpose.